### PR TITLE
[2201.9.0] Add build tool documentation

### PR DESCRIPTION
**Pull request by gayaldassanayake**
## Purpose
$subject
Related issues - https://github.com/ballerina-platform/ballerina-lang/issues/42155

## Checklist

- [ ] **Page addition**
  - [ ] Add `permalink` to pages.

- [ ] **Page removal**
  - [ ] Remove entry from corresponding left nav YAML file.
  - [ ] Add `redirect_from` on the alternative page.
  - [ ] If no alternative page, add redirection on the `redirections.js ` file.

- [ ] **Page rename**
  - [ ] Add front-matter `redirect_from`.
  - [ ] Add front-matter `redirect_to:` (if applicable).

- [ ] **Page restrcuture**
  - [ ] Add `permalink` to pages.
  - [ ] Add front-matter `redirect_from`.
  - [ ] Add front-matter `redirect_to:` (if applicable).

### CODE REVIEW

File: swan-lake/development-tutorials/source-code-dependencies/package-references.md [hunk 1]
```diff
@@ -349,6 +349,48 @@ The following example shows a platform dependency entry with the `scope`.
   scope =  "<scope-of-the-jar-file>"
   ```
 
+### Tools
+
+You can specify code generation tools to integrate with the package build. These tools execute before the package build and generate code used during the build process.
```
Comment (azinneera) [inline swan-lake/development-tutorials/source-code-dependencies/package-references.md:354]:
```suggestion
You can specify code generation tools to integrate with the package build. These tools execute before the package build and generate code that is essential for the build process.
```

File: swan-lake/development-tutorials/source-code-dependencies/package-references.md [hunk 2]
```diff
@@ -349,6 +349,48 @@ The following example shows a platform dependency entry with the `scope`.
...
+
+The following example shows how to specify a tool in the `Ballerina.toml` file.
+
+```toml
+[[tool.<command>]]
+id = "<tool-id>"
+filePath = "<specification-file>"
+targetModule = "<generated-code-destination-module>"
```
Comment (azinneera) [inline swan-lake/development-tutorials/source-code-dependencies/package-references.md:362]:
```suggestion
targetModule = "<destination-module-to-generated-code->"
```

File: swan-lake/development-tutorials/source-code-dependencies/package-references.md [hunk 3]
```diff
@@ -349,6 +349,48 @@ The following example shows a platform dependency entry with the `scope`.
...
+### Tools
+
+You can specify code generation tools to integrate with the package build. These tools execute before the package build and generate code used during the build process.
+
+The following example shows how to specify a tool in the `Ballerina.toml` file.
+
+```toml
+[[tool.<command>]]
+id = "<tool-id>"
+filePath = "<specification-file>"
```
Comment (azinneera) [inline swan-lake/development-tutorials/source-code-dependencies/package-references.md:361]:
```suggestion
filePath = "<schema-or-API-specification-file>"
```

File: swan-lake/development-tutorials/source-code-dependencies/package-references.md [hunk 4]
```diff
@@ -349,6 +349,48 @@ The following example shows a platform dependency entry with the `scope`.
...
+targetModule = "<generated-code-destination-module>"
+options.<option1> = "<value1>"
+options.<option2> = "<value2>"
+```
+
+The tool command that you need to use should be specified after the `tool.` prefix in the table array header.
+
+The mandatory `id` field specifies a unique identifier for the tool entry, as there can be multiple entries for the same tool command to generate code for different modules. The `id` must consist of alphanumeric characters and underscores only, but it must not begin or end with an underscore, and consecutive underscores are not permitted.
```
Comment (azinneera) [inline swan-lake/development-tutorials/source-code-dependencies/package-references.md:369]:
```suggestion
The mandatory `id` field specifies a unique identifier for the tool entry, as a tool can utilize multiple schemas/API specifications files. The `id` must consist of alphanumeric characters and underscores only, and must not begin or end with an underscore. Consecutive underscores are also not permitted.
```

File: swan-lake/development-tutorials/source-code-dependencies/package-references.md [hunk 5]
```diff
@@ -349,6 +349,48 @@ The following example shows a platform dependency entry with the `scope`.
...
+
+The `filePath` field is mandatory, providing the path to the specification file that the tool uses to generate code.
+ 
+The `targetModule` field specifies the module where the generated code should be placed. If this is not specified, it will default to the root module. This should be unique for each tool entry.
+ 
+The `options` fields are defined by the tool developer and used to pass additional parameters to the tool.
```
Comment (azinneera) [inline swan-lake/development-tutorials/source-code-dependencies/package-references.md:375]:
```suggestion
The `options` fields can be used to pass additional parameters to the tool.
```

File: swan-lake/development-tutorials/source-code-dependencies/package-references.md [hunk 6]
```diff
@@ -349,6 +349,48 @@ The following example shows a platform dependency entry with the `scope`.
...
+
+If a tool contains multiple subcommands, you can specify them as follows.
```
Comment (azinneera) [inline swan-lake/development-tutorials/source-code-dependencies/package-references.md:377]:
```suggestion
If a tool provides multiple subcommands, you can specify them as follows.
```

File: swan-lake/development-tutorials/source-code-dependencies/package-references.md (hunk 2 above)
Comment (gayaldassanayake) [reply swan-lake/development-tutorials/source-code-dependencies/package-references.md]: Fixed with 8ca99fb230b7749756c6b0f36bdd01f05cef3708

File: swan-lake/development-tutorials/source-code-dependencies/package-references.md (hunk 3 above)
Comment (gayaldassanayake) [reply swan-lake/development-tutorials/source-code-dependencies/package-references.md]: Fixed with 8ca99fb230b7749756c6b0f36bdd01f05cef3708

File: swan-lake/development-tutorials/source-code-dependencies/package-references.md (hunk 4 above)
Comment (gayaldassanayake) [reply swan-lake/development-tutorials/source-code-dependencies/package-references.md]: Fixed with 8ca99fb230b7749756c6b0f36bdd01f05cef3708

File: swan-lake/development-tutorials/source-code-dependencies/package-references.md (hunk 5 above)
Comment (gayaldassanayake) [reply swan-lake/development-tutorials/source-code-dependencies/package-references.md]: Fixed with 8ca99fb230b7749756c6b0f36bdd01f05cef3708

File: swan-lake/development-tutorials/source-code-dependencies/package-references.md (hunk 6 above)
Comment (gayaldassanayake) [reply swan-lake/development-tutorials/source-code-dependencies/package-references.md]: Fixed with 8ca99fb230b7749756c6b0f36bdd01f05cef3708

File: swan-lake/development-tutorials/source-code-dependencies/package-references.md (hunk 1 above)
Comment (gayaldassanayake) [reply swan-lake/development-tutorials/source-code-dependencies/package-references.md]: Fixed with 8ca99fb230b7749756c6b0f36bdd01f05cef3708

**Comment by sm1990** [comment]
LGTM

**Review by azinneera — APPROVED** [review]

**Review by sm1990 — APPROVED** [review]

**Pull request merged**